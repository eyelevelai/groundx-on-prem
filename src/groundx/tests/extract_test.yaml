suite: "extract renders as expected"
chart:
  appVersion: 0.1.0
  version: 0.1.0

tests:
  - it: "default: extract"
    templates:
      - ../templates/resources/extract-config-py.yaml
    values:
      - ./files/values.extract.yaml
    asserts:
      - matchSnapshot: {}
  - it: "default: extract.microservice"
    templates:
      - ../templates/resources/extract-config-py.yaml
    values:
      - ./files/values.extract.microservice.yaml
    asserts:
      - matchSnapshot: {}
  - it: "default: extract.existing.1"
    set:
      extract.file.url: http://s3.com
    templates:
      - ../templates/resources/extract-config-py.yaml
    values:
      - ./files/values.extract.microservice.yaml
    asserts:
      - matchSnapshot: {}
