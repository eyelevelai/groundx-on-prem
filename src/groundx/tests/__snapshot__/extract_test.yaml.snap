'default: extract':
  1: |
    apiVersion: v1
    data:
      config.py: |
        from classes.settings import (
            AgentSettings,
            ContainerSettings,
            ContainerUploadSettings,
        )

        agent_settings = AgentSettings(
            api_base="http://summary-api.eyelevel.svc.cluster.local",
            api_key_env="GROUNDX_AGENT_API_KEY",
            model_id="google/gemma-3-4b-it",
        )

        container_settings = ContainerSettings(
            broker="rediss://X.X.X.use2.cache.amazonaws.com:6379/0",
            callback_api_key="00000000-0000-0000-0000-000000000000",
            metrics_broker="rediss://X.X.X.use2.cache.amazonaws.com:6379/0",
            service="extract",
            upload=ContainerUploadSettings(
                base_domain="X.s3.us-west-2.amazonaws.com",
                bucket="eyelevel",
                region="us-west-2",
                ssl=True,
                type="s3",
                url="https://X.s3.us-west-2.amazonaws.com/",
                key="awsKey",
                secret="awsSecret",
            ),
            valid_api_keys=[
              "00000000-0000-0000-0000-000000000000",
              "00000000-0000-0000-0000-000000000000",
            ],
            workers=1,
        )
    kind: ConfigMap
    metadata:
      labels:
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        version: 0.1.0
      name: extract-config-py-map
      namespace: eyelevel
'default: extract.existing.1':
  1: |
    apiVersion: v1
    data:
      config.py: |
        from classes.settings import (
            AgentSettings,
            ContainerSettings,
            ContainerUploadSettings,
        )

        agent_settings = AgentSettings(
            api_key_env="GROUNDX_AGENT_API_KEY",
        )

        container_settings = ContainerSettings(
            broker="redis://cache.eyelevel.svc.cluster.local:6379/0",
            callback_api_key="00000000-0000-0000-0000-000000000001",
            google_sheets_drive_id="abc",
            google_sheets_template_id="a123",
            metrics_broker="redis://cache-metrics.eyelevel.svc.cluster.local:6379/0",
            service="extract",
            upload=ContainerUploadSettings(
                base_domain="s3.com",
                bucket="test",
                region="us-east-1",
                ssl=False,
                type="s3",
                url="http://s3.com/",
                key="key",
                secret="pass",
            ),
            valid_api_keys=[
              "00000000-0000-0000-0000-000000000000",
              "00000000-0000-0000-0000-000000000000",
              "00000000-0000-0000-0000-000000000002",
            ],
            workers=1,
        )
    kind: ConfigMap
    metadata:
      labels:
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        version: 0.1.0
      name: extract-config-py-map
      namespace: eyelevel
'default: extract.microservice':
  1: |
    apiVersion: v1
    data:
      config.py: |
        from classes.settings import (
            AgentSettings,
            ContainerSettings,
            ContainerUploadSettings,
        )

        agent_settings = AgentSettings(
            api_key_env="GROUNDX_AGENT_API_KEY",
        )

        container_settings = ContainerSettings(
            broker="redis://cache.eyelevel.svc.cluster.local:6379/0",
            callback_api_key="00000000-0000-0000-0000-000000000001",
            google_sheets_drive_id="abc",
            google_sheets_template_id="a123",
            metrics_broker="redis://cache-metrics.eyelevel.svc.cluster.local:6379/0",
            service="extract",
            upload=ContainerUploadSettings(
                base_domain="X.s3.us-east-1.amazonaws.com",
                bucket="test",
                region="us-east-1",
                ssl=True,
                type="s3",
                url="https://X.s3.us-east-1.amazonaws.com/",
                key="key",
                secret="pass",
            ),
            valid_api_keys=[
              "00000000-0000-0000-0000-000000000000",
              "00000000-0000-0000-0000-000000000000",
              "00000000-0000-0000-0000-000000000002",
            ],
            workers=1,
        )
    kind: ConfigMap
    metadata:
      labels:
        appVersion: 0.1.0
        chart: groundx-0.1.0
        heritage: Helm
        version: 0.1.0
      name: extract-config-py-map
      namespace: eyelevel
